src/style/sky.ts

```diff
@@ -66,6 +66,7 @@ export class Sky extends Evented {
                 'sky-color': 'transparent',
                 'horizon-color': 'transparent',
                 'fog-color': 'transparent',
+                'fog-ground-blend': 1
```

Comment:
```suggestion
                'fog-ground-blend': 1,
                'atmosphere-blend': 0,
```

This will also disable atmosphere by default on globe, and remove the need for the `setSky` calls in globe examples. Also `map.setSky(undefined);` should work with this change.